### ISSUE TITLE
markdown inside inline block-level HTML?

### ISSUE THREAD

**Issue by bollwyvl** [jupyter/nbconvert#45]
The live Notebook allows for rendering markdown inside inline block-level html, such that both of these show up in the notebook:

``` markdown
<div>[a link in a block](#)</div>
<span>[a link in a span](#)</span>
```

nbconvert, and therefore nbviewer, will not render the first link if using the mistune engine, as the author has indicated that mistune will continue to be spec-driven in lepture/mistune#51, following [this guidance](http://daringfireball.net/projects/markdown/syntax#html):

> Note that Markdown formatting syntax is not processed within block-level HTML tags. E.g., you can’t use Markdown-style _emphasis_ inside an HTML block.
> ...
> Unlike block-level HTML tags, Markdown syntax _is_ processed within span-level tags.

So... options:
- this repo could opine "more like Notebook", and choose to re-implement this on top of mistune (we're already overloading it for math) or,
- the Notebook could opine "more like the spec" and turn on marked's `pedantic` mode, potentially breaking a lot of notebooks for being not up to spec.
- use [commonmark](http://commonmark.org/) and break everything

> linked to  jupyter/nbviewer#458

**Comment by lepture** [comment]
Personally, I think we shouldn't process text in any html tags. But it is easy to change the behaviour with options.

Here is what I did in the recent commits:

1. configure with **parse_html=True**, text in any html tags will be parsed as markdown
2. configure with **parse_inline_html=True**, text in inline tags will be parsed
3. configure with **parse_block_html=True**, text in block tags will be parsed

**Comment by bollwyvl** [comment]
Great! I usually come down on the side of spec adherence, but the way we
used marked has set expectations otherwise.

Thanks, hopefully it should be a one liner pr to nbconvert and backport to
3.x!

**Comment by michaelpacer** [comment]
@bollwyvl @lepture In those commits did you address this problem? Can this issue be closed as is (or should it be assigned to the 5.0 milestone if it hasn't been addressed as part of nbconvert proper)?

**Comment by MSeal** [comment]
Closing issue. Comment or reopen if this is still an issue.
